packages/core/src/state-management/named-p-queue.ts

```diff
@@ -0,0 +1,63 @@
+import PQueue from 'p-queue';
+import { Mutex } from 'await-semaphore';
+
+/**
+ * Set of named PQueues.
+ * When a task is done, it checks for pending tasks. No pending tasks means PQueue for the name is cleared.
+ */
+export class NamedPQueue {
+  #lanes: Map<string, PQueue>;
+  #operations: Mutex = new Mutex();
+
+  constructor(lanes: Map<string, PQueue> = new Map()) {
```

Comment:
is there ever a reason you wouldn't want to start with an empty map?  Can we remove this argument from the constructor?

Reply:
Testing is the reason. Either this or exposing lanes to the outside.